test/both.js

```diff
@@ -38,4 +38,18 @@ describe('both', function() {
     eq(evenAnd(gt10)(12), true);
   });
 
+  it('accepts fantasy-land applicative functors', function() {
+    function Just(x) { this.value = x; }
+    Just.prototype.map = function(f) { return new Just(f(this.value)); };
+    Just.prototype.ap = function(maybe) { return maybe.map(this.value); };
+    function Nothing() {}
+    Nothing.prototype.map = function() { return this; };
+    Nothing.prototype.ap = function() { return this; };
+    eq(R.both(new Just(true), new Just(true)).value, true);
+    eq(R.both(new Just(true), new Just(false)).value, false);
+    eq(R.both(new Just(true), new Nothing()) instanceof Nothing, true);
+    eq(R.both(new Nothing(), new Just(false)) instanceof Nothing, true);
+    eq(R.both(new Nothing(), new Nothing()) instanceof Nothing, true);
+  });
+
```

Comment:
Sanctuary is a dev dependency. Let's use it here. Not only will it obviate the need to define `Just` and `Nothing`, but the assertions will be nicer to read:

```diff
-    eq(R.both(new Just(true), new Just(true)).value, true);
-    eq(R.both(new Just(true), new Just(false)).value, false);
-    eq(R.both(new Just(true), new Nothing()) instanceof Nothing, true);
-    eq(R.both(new Nothing(), new Just(false)) instanceof Nothing, true);
-    eq(R.both(new Nothing(), new Nothing()), Nothing());
+    eq(R.both(Just(true), Just(true)), Just(true));
+    eq(R.both(Just(true), Just(false)), Just(false));
+    eq(R.both(Just(true), Nothing()), Nothing());
+    eq(R.both(Nothing(), Just(false)), Nothing());
+    eq(R.both(Nothing(), Nothing()), Nothing());
```